crates/cubecl-core/src/runtime_tests/plane.rs

```diff
@@ -75,132 +74,296 @@ pub fn kernel_broadcast<F: Float>(output: &mut Tensor<F>) {
     }
 }
 
-pub fn test_plane_sum<TestRuntime: Runtime, F: Float + CubeElement>(
+pub fn test_plane_sum<TestRuntime: Runtime, F: Float + CubeElement + Sized>(
     client: ComputeClient<TestRuntime::Server, TestRuntime::Channel>,
+    vectorization: u8,
 ) {
+    let plane_size = 32;
+    let input: Vec<f32> = (0..plane_size * vectorization as u32)
+        .map(|x| x as f32)
+        .collect();
+    let mut expected = input.clone();
+
+    for k in 1..plane_size as usize {
+        for v in 0..vectorization as usize {
+            expected[v] += input[v + k * vectorization as usize];
+        }
+    }
+    let input: Vec<F> = input.into_iter().map(|x| F::new(x)).collect();
+    let expected: Vec<F> = expected.into_iter().map(|x| F::new(x)).collect();
+
     test_plane_operation::<TestRuntime, F, _>(
-        as_type![F: 4.0, 5.0, 7.0, 1.0],
-        as_type![F: 17.0, 5.0, 7.0, 1.0],
+        &input,
+        &expected,
+        vectorization,
         client.clone(),
-        |cube_count: CubeCount, cube_dim, handle| {
-            kernel_sum::launch::<F, TestRuntime>(&client, cube_count, cube_dim, handle)
+        |cube_count, handle| {
+            kernel_sum::launch::<F, TestRuntime>(
+                &client,
+                cube_count,
+                CubeDim::new(32, 1, 1),
+                handle,
+            )
         },
     );
 }
 
 pub fn test_plane_prod<TestRuntime: Runtime, F: Float + CubeElement>(
     client: ComputeClient<TestRuntime::Server, TestRuntime::Channel>,
+    vectorization: u8,
 ) {
+    let plane_size = 32;
+    let input: Vec<f32> = (0..plane_size * vectorization as u32)
+        .map(|x| match x % 3 {
+            0 => 0.5,
```

Comment:
A bit picky on my part, but I think you should pick other values. 0.5 and 2. will cancel each other, while 1.0 doesn't affect the result. Therefore it would be possible to arrive at the right result coincidentally

Reply:
changed to 0.75, 1.25 and 1.75 (I want number that are exactly representable to avoid floating-point errors)